databases/week1/lesson1_homework.sql

```diff
@@ -0,0 +1,56 @@
+-- 1. Find out how many tasks are in the task table
+SELECT count(id)
+FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT count(id) AS tasks_with_invalid_due_date
+FROM task
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE status_id = 3;
+
+-- 4. Find all the tasks that are not marked as done
+SELECT *
+FROM task
+WHERE status_id != 3
+ORDER BY status_id;
+
+-- 5. Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM task
+ORDER BY created DESC;
+
+-- 6. Get the single most recently created task
+SELECT *
+FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date
+FROM task
+WHERE title REGEXP 'database' OR description REGEXP 'database';
+
+-- 8. Get the title and status (as text) of all tasks
+SELECT task.title, status.name AS status_description
+FROM task
+INNER JOIN status
+ORDER BY status_description;
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT status.name, COUNT(task.status_id) AS 'number_of_task'
+FROM task
+INNER JOIN status
+ON task.status_id = status.id
+GROUP BY task.status_id;
```

Comment:
The `group by` should be by `status.name`
```suggestion
GROUP BY status.name;
```